File: src/main/java/org/revcloud/vader/matchers/IntMatchers.java
package org.revcloud.vader.matchers;

import lombok.experimental.UtilityClass;
import org.hamcrest.Matcher;

import static org.hamcrest.Matchers.allOf;
import static org.hamcrest.Matchers.greaterThanOrEqualTo;
import static org.hamcrest.Matchers.lessThanOrEqualTo;

@UtilityClass
public class IntMatchers {
    public static Matcher<Integer> inRangeInclusive(int start, int end) {
        return allOf(greaterThanOrEqualTo(start), lessThanOrEqualTo(end));
    }
}

File: src/main/java/org/revcloud/vader/matchers/IntMatchers.kt
@file:JvmName("IntMatchers")

package org.revcloud.vader.matchers

import org.hamcrest.Matcher
import org.hamcrest.Matchers.allOf
import org.hamcrest.Matchers.greaterThanOrEqualTo
import org.hamcrest.Matchers.lessThanOrEqualTo

fun inRangeInclusive(start: Int, end: Int): Matcher<Int> =
  allOf(greaterThanOrEqualTo(start), lessThanOrEqualTo(end))
